nixos/modules/services/misc/matrix-appservice-irc.nix

```diff
@@ -0,0 +1,123 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  dataDir = "/var/lib/matrix-appservice-irc";
+in {
+  options = {
+    services.matrix-appservice-irc = {
+      enable = mkEnableOption "a Node.js IRC bridge for Matrix";
+
+      port = mkOption {
+        type = types.port;
+        default = 9999; # from https://github.com/matrix-org/matrix-appservice-irc/blob/0.12.0/README.md#4-running
+        description = ''
+          Port number on which the bridge should listen for internal communication with the Matrix homeserver.
+        '';
+      };
+
+      config = mkOption rec {
+        type = types.attrs;
+        apply = recursiveUpdate default;
+        default = {
+          ircService = {
+            databaseUri = "nedb://${dataDir}/data";
+            passwordEncryptionKeyPath = "${dataDir}/passkey.pem";
+          };
+        };
+        example = literalExample ''
+          {
+            homeserver = {
+              url = "http://localhost:8008";
+              domain = "localhost";
+            };
+            ircService.servers = {
+              "irc.example.com" = {
+                name = "ExampleNet";
+                dynamicChannels.groupId = "+myircnetwork:localhost";
+              };
+            };
+          }
+        '';
+        description = ''
+          <filename>config.yaml</filename> configuration as a Nix attribute set.
+
+          Configuration options should match those described in
+          <link xlink:href="https://github.com/matrix-org/matrix-appservice-irc/blob/0.12.0/config.sample.yaml">
+          config.sample.yaml</link>.
+
+          Secret tokens should be specified using <option>environmentFile</option>
+          instead of this world-readable attribute set.
+        '';
+      };
+
+      localpart = mkOption {
+        type = types.string;
```

Comment:
`types.string` is deprecated (it concats multiple values of `localpart` is set by multiple modules) similar to `types.lines` (which concats with `\n`), `types.str` should be used as it allows only one value (that can be forcefully modified with `mkForce`).

Reply:
Amended.